processor/stream/result.go

```diff
@@ -18,94 +18,73 @@
 package stream
 
 import (
-	"fmt"
+	"net/http"
 	"strings"
 
-	"github.com/elastic/beats/libbeat/monitoring"
-)
-
-type Error struct {
-	Type     StreamError `json:"-"`
-	Message  string      `json:"message"`
-	Document string      `json:"document,omitempty"`
-}
-
-func (s *Error) Error() string {
-	if s.Document != "" {
-		return fmt.Sprintf("%s [%s]", s.Message, string(s.Document))
-	}
-	return s.Message
-}
-
-type StreamError int
+	"github.com/elastic/apm-server/server"
 
-const (
-	QueueFullErrType StreamError = iota
-	InvalidInputErrType
-	InputTooLargeErrType
-	ShuttingDownErrType
-	ServerErrType
-	MethodForbiddenErrType
-	RateLimitErrType
+	"github.com/elastic/beats/libbeat/monitoring"
 )
 
 const (
 	errorsLimit = 5
 )
 
 var (
-	m             = monitoring.Default.NewRegistry("apm-server.processor.stream")
-	mAccepted     = monitoring.NewInt(m, "accepted")
-	monitoringMap = map[StreamError]*monitoring.Int{
-		QueueFullErrType:     monitoring.NewInt(m, "errors.queue"),
-		InvalidInputErrType:  monitoring.NewInt(m, "errors.invalid"),
-		InputTooLargeErrType: monitoring.NewInt(m, "errors.toolarge"),
-		ShuttingDownErrType:  monitoring.NewInt(m, "errors.server"),
-		ServerErrType:        monitoring.NewInt(m, "errors.closed"),
```

Comment:
I agree that the terminating error types don't add a lot of value as they are reflected in the response counters from the handler. 
Just wanted to ensure removing the counters for invalid and too large errors was also intentional. They show how many invalid events have been sent up, which cannot be seen otherwise afaics. 
How about still counting those errors here and returning them in the result so the handler in the beater package can log them?

Reply:
Good point, will give it another thought.